test/ConfigurationProvider.test.js

```diff
@@ -0,0 +1,251 @@
+ *
+ * Copyright © 2023 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    FileConfigurationProvider,
+    FolderConfigurationProvider
+} from "../src/config/ConfigurationProvider.js";
+
+import fs from "node:fs/promises";
+import os from "node:os";
+import path from "node:path";
+
+const jsonConfig = `{
+    "name": "configuration-provider-test-json-config",
+    "plugins": [
+        "i18nlint-plugin-test"
+    ],
+    "locales": [
+        "en-US",
+        "de-DE"
+    ]
+}`;
+
+const cjsConfig = `module.exports = {
+    "name": "configuration-provider-test-cjs-config",
+    "plugins": [
+        "i18nlint-plugin-test"
+    ],
+    "locales": [
+        "en-US",
+        "de-DE"
+    ]
+};`;
+
+const esmConfig = `export default {
+    "name": "configuration-provider-test-esm-config",
+    "plugins": [
+        "i18nlint-plugin-test"
+    ],
+    "locales": [
+        "en-US",
+        "de-DE"
+    ]
+};`;
+
+const esmPackage = `{
+    "type": "module"
+}`;
+
+describe("Configuration Provider", () => {
+    let tempDir;
+    beforeEach(async () => {
+        tempDir = await fs.mkdtemp(path.join(os.tmpdir(), "i18nlint-test-"));
+    });
+    afterEach(async () => {
+        await fs.rm(tempDir, { recursive: true });
+    });
+    describe("File Configuration Provider", () => {
+        test("load config from JSON file", async () => {
+            // should successfully import the JSON config file
+
+            // setup - write config to file
+            const configPath = path.join(tempDir, `ilib-lint-config.json`);
+            await fs.writeFile(configPath, jsonConfig);
+
+            // test
+            const provider = new FileConfigurationProvider(configPath);
+            const result = await provider.loadConfiguration();
+
+            expect(result.name).toBe("configuration-provider-test-json-config");
+        });
+
+        test("load config from CJS file", async () => {
+            // should successfully import the JS config file from a CommonJS package
+
+            // write config to file
+            const configPath = path.join(tempDir, `ilib-lint-config.js`);
+            await fs.writeFile(configPath, cjsConfig);
+
+            // test
+            const provider = new FileConfigurationProvider(configPath);
+            const result = await provider.loadConfiguration();
+
+            expect(result.name).toBe("configuration-provider-test-cjs-config");
+        });
+
```

Comment:
What happens if the user provides a .js file that is in commonjs format instead of ESM?

Reply:
As long as the package is also CJS, all will be ok. Otherwise, `import()` call will throw

Reply:
added tests for cross loading without extensions in a4736bd